deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -974,16 +974,27 @@ def evaluate_generator(self,
 
   def get_layer_variables(self, layer):
     """Get the list of trainable variables in a layer of the graph."""
-    if tfe.in_eager_mode():
-      return layer.variables
     if not self.built:
       self.build()
     with self._get_tf("Graph").as_default():
+      if tfe.in_eager_mode():
```

Comment:
@peastman: A couple of remarks:
1. I moved the `self.build()` condition to the top, otherwise the eager mode was returning only empty lists.

2. The test for eager mode, previously threw up an error `ValueError(Attempted to read a variable from another isolation environment or Graph)`. Had to use the `self._get_tf("Graph").as_default()` condition before checking for eager mode to make sure the graphs were the same. The same error also occured for `self.get_layer_variables`, so added it there as well.
I think this is the correct way, what do you think?

Reply:
Those both sound reasonable.